### ISSUE TITLE
`describe` command should fetch all the information possible and display failures in the end

### ISSUE THREAD

**Issue by joaopapereira** [carvel-dev/imgpkg#347]
**Describe the problem/challenge you have**
Sometimes when trying to get more information about a Bundle an error can occur while talking with the registry or the user might not have access to some particular Image. Today, when `imgpkg` fails to retrieve information about some image it displays the error message and stops processing. Nevertheless, `imgpkg` could still provide information about the rest of the Images that are part of the Bundle. 

**Describe the solution you'd like**

When running the command `imgpkg describe` the command should run until the end without falling and in the end provide information about the failures

```
Bundle SHA: sha256:3cd8f0d917a92fafeef2d44eb63195d5351c35cf46b8b7448cb126c7d68c6291

Images:
  - Image: localhost:5000/test@sha256:4c8b96d4fffdfae29258d94a22ae4ad1fe36139d47288b8960d9958d1e63a9d0
    Type: Image
    Origin: index.docker.io/dkalinin/k8s-simple-app@sha256:4c8b96d4fffdfae29258d94a22ae4ad1fe36139d47288b8960d9958d1e63a9d0
    Annotations:
      kbld.carvel.dev/id: index.docker.io/dkalinin/k8s-simple-app:latest
      kbld.carvel.dev/origins: - resolved:
    tag: latest
    url: index.docker.io/dkalinin/k8s-simple-app:latest


  - Image: localhost:5000/test@sha256:510482df49db421542ae10142e8ce99572b23ef72675c37e92e4a12b541a3f6a
    Type: Internal

  - Image: localhost:5000/test@sha256:ecc10da4f480b1ba263d96197151d76bc787104cbcf7f1b1f42b14340b58bae7
    Type: Signature
    Error: Access Denied

Unable to describe the full bundle, please review the previous output to understand the error
Failed
```

When the error happens while reading an Image from ImagesLock, `imgpkg` should assume that the type of the Image is `Image` and should provide the following output

```
    Image: localhost:5000/test@sha256:4c8b96d4fffdfae29258d94a22ae4ad1fe36139d47288b8960d9958d1e63a9d0
    Type: Image
    Origin: index.docker.io/dkalinin/k8s-simple-app@sha256:4c8b96d4fffdfae29258d94a22ae4ad1fe36139d47288b8960d9958d1e63a9d0
    Error: Image Not Found
```

**Anything else you would like to add:**
For signatures/sboms/internal Images we should ignore 404 errors since they might not be present, nevertheless, if we get a different error it should be logged.
This behavior should apply to all the output types.

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help work on this issue.

**Comment by joaopapereira** [comment]
Since this has been referenced multiple times since the creation of the feature, I am bumping it to Prioritized Backlog to be tackled in the near future

**Comment by joe-kimmel-vmw** [comment]
@joaopapereira just poking at this a tiny bit --   do you think the work starts at this line: https://github.com/vmware-tanzu/carvel-imgpkg/blob/develop/pkg/imgpkg/v1/describe.go#L100

or am i not yet reading the right part even?

**Comment by joaopapereira** [comment]
This is a 2 part change and I believe the implementation will have to happen inside `FetchAllImagesRefs` because it is inside that function that we [retrieve signatures](https://github.com/vmware-tanzu/carvel-imgpkg/blob/d5c9d0c9a8c62293f371fcd7b8d772dab215f795/pkg/imgpkg/bundle/fetch_images.go#L32) and we also have a [recursive function](https://github.com/vmware-tanzu/carvel-imgpkg/blob/d5c9d0c9a8c62293f371fcd7b8d772dab215f795/pkg/imgpkg/bundle/bundle_images_lock.go#L85) to get all the bundle images that we need to make sure it returns all the information it could grab in case of error.


So as a plan of attack you can think of these to places as the source of the information we want to display

**Comment by aaronshurley** [comment]
Shipped as a part of [v0.34.0](https://github.com/vmware-tanzu/carvel-imgpkg/releases/tag/v0.34.0).
